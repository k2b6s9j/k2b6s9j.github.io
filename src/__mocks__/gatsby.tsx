import * as React from "react";
const gatsby = jest.requireActual("gatsby");
import { graphql } from "graphql";

type MockedGraphQLReturn = Promise<any>;

const mockedGraphQLQuery = (query: string): MockedGraphQLReturn =>
  graphql(undefined, query);

interface StaticQueryHOCProps {
  query: MockedGraphQLReturn;
  render?: (data: object) => React.ReactElement;
  children?: (data: object) => React.ReactElement;
}

const mockedStaticQueryHOC = (
  props: StaticQueryHOCProps
): React.ReactElement => {
  if (!props.children || !props.render) {
    throw new Error("Render prop must be provided!");
  }

  return (
    (props.children && props.children({})) ||
    (props.render && props.render({})) || <div>Error!</div>
  );
};

const mockedStaticQueryHook = (data: MockedGraphQLReturn): object => ({});

module.exports = {
  ...gatsby,
  graphql: jest.fn(mockedGraphQLQuery),
  Link: jest.fn(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      to,
      ...rest
    }): React.ReactElement => <a {...rest} href={to} />
  ),
  StaticQuery: jest.fn(mockedStaticQueryHOC),
  useStaticQuery: jest.fn(mockedStaticQueryHook)
};
