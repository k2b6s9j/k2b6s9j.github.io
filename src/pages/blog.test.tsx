import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import { Helmet } from "react-helmet";
import Page from "./blog";

beforeEach((): void => {
  (useStaticQuery as jest.Mock).mockImplementation((): object => ({
    file: {
      childImageSharp: {
        fixed: {
          base64: "",
          width: 100,
          height: 100,
          src: "",
          srcSet: ""
        }
      }
    },
    backdrop: {
      childImageSharp: {
        fluid: {
          base64: "",
          aspectRatio: 1.333,
          src: "",
          srcSet: "",
          srcWebp: "",
          srcSetWebp: "",
          sizes: ""
        }
      }
    },
    site: {
      siteMetadata: {
        title: "",
        description: "",
        siteUrl: "",
        twitterUsername: "",
        instagramUsername: "",
        linkedinUsername: "",
        githubUsername: ""
      }
    }
  }));
});

describe("Blog Page", (): void => {
  it("renders correctly", (): void => {
    const tree = renderer
      .create(
        <Page
          data={{
            allMarkdownRemark: {
              edges: [
                {
                  node: {
                    id: "",
                    rawMarkdownBody: "",
                    excerpt: "",
                    fields: {
                      slug: ""
                    },
                    frontmatter: {
                      title: "",
                      date: "",
                      isoDate: "",
                      description: ""
                    },
                    timeToRead: "",
                    wordCount: {
                      words: ""
                    }
                  }
                }
              ]
            }
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();

    // Get a snapshot of the head since this component modifies it.
    const helmet = Helmet.peek();
    expect(helmet).toMatchSnapshot();
  });
});
