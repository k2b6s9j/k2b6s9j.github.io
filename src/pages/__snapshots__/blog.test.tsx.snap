// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Page renders correctly 1`] = `
Array [
  .emotion-7 {
  height: 3rem;
  width: 100vw;
  margin-bottom: 1em;
}

.emotion-6 {
  height: 3rem;
  max-width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1rem;
  padding-right: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-1 {
  text-align: start;
  max-width: 50%;
}

.emotion-0 {
  font-weight: 600;
  color: #141414;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-5 {
  text-align: end;
  max-width: 50%;
}

.emotion-5 > :not(:last-child) {
  margin-right: 1em;
}

<div
    className="emotion-7 gatsby-image-wrapper"
    style={
      Object {
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "opacity": 0.99,
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
          .emotion-7:before,
          .emotion-7:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 0.25s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .emotion-7:before {
            z-index: -100;
            
            
            opacity: 0; 
          }
          .emotion-7:after {
            z-index: -101;
            
            
          }
        ",
        }
      }
    />
    <noscript>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
          .emotion-7:before {
            opacity: 1;
            background-image: ;
          }",
          }
        }
      />
    </noscript>
    <nav
      className="emotion-6"
    >
      <div
        className="emotion-1"
      >
        <a
          className="emotion-0"
          href="/"
        >
          Kepler Sticka-Jones
        </a>
      </div>
      <div
        className="emotion-5"
      >
        <a
          className="emotion-0"
          href="/blog"
        >
          Blog
        </a>
        <a
          className="emotion-0"
          href="/projects"
        >
          Projects
        </a>
        <a
          className="emotion-0"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </nav>
  </div>,
  .emotion-1 {
  max-width: 55em;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:55em) {
  .emotion-1 {
    margin-left: 2em;
    margin-right: 2em;
  }
}

.emotion-0:not(:last-child) {
  margin-bottom: 1.5rem;
}

<div
    className="emotion-1"
  >
    <h1>
      Blog
    </h1>
    <div>
      <div
        className="emotion-0"
      >
        <a
          href=""
        >
          <h2>
            
          </h2>
        </a>
        <div>
          <span>
            Published 
            
          </span>
          <span>
             | 
          </span>
          <span>
            
             words
          </span>
          <span>
             | 
          </span>
          <span>
            
             minute read
          </span>
        </div>
        <div>
          
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Blog Page renders correctly 2`] = `
Object {
  "baseTag": Array [],
  "bodyAttributes": Object {},
  "defer": true,
  "encode": true,
  "htmlAttributes": Object {
    "lang": "en",
  },
  "linkTags": Array [],
  "metaTags": Array [
    Object {
      "content": "Kepler Sticka-Jones",
      "name": "author",
    },
    Object {
      "content": "",
      "name": "description",
    },
    Object {
      "content": "#409040",
      "name": "theme-color",
    },
    Object {
      "content": "Blog | ",
      "property": "og:title",
    },
    Object {
      "content": "website",
      "property": "og:type",
    },
    Object {
      "content": "",
      "property": "og:url",
    },
    Object {
      "content": "",
      "property": "og:image",
    },
    Object {
      "content": "image/jpeg",
      "property": "og:image:type",
    },
    Object {
      "content": "480",
      "property": "og:image:width",
    },
    Object {
      "content": "480",
      "property": "og:image:height",
    },
    Object {
      "content": "",
      "property": "og:description",
    },
    Object {
      "content": "summary",
      "name": "twitter:card",
    },
    Object {
      "content": "@",
      "name": "twitter:site",
    },
    Object {
      "content": "@",
      "name": "twitter:creator",
    },
    Object {
      "content": "E5D802A305BE2DD99EB4484A4B1DD90E",
      "name": "msvalidate.01",
    },
  ],
  "noscriptTags": Array [],
  "onChangeClientState": [Function],
  "scriptTags": Array [
    Object {
      "innerHTML": "{\\"@context\\":\\"http://www.schema.org\\",\\"@type\\":\\"CollectionPage\\",\\"name\\":\\"Blog | Kepler Sticka-Jones\\",\\"url\\":\\"/blog\\",\\"about\\":{\\"@type\\":\\"Blog\\",\\"url\\":\\"/blog\\",\\"blogPosts\\":[{\\"@type\\":\\"BlogPosting\\",\\"url\\":\\"\\",\\"name\\":\\"\\",\\"headline\\":\\"\\",\\"wordCount\\":\\"\\",\\"description\\":\\"\\",\\"author\\":{\\"@type\\":\\"Person\\",\\"name\\":\\"Kepler Sticka-Jones\\",\\"url\\":\\"https://keplersj.com\\"},\\"publisher\\":{\\"@type\\":\\"Person\\",\\"name\\":\\"Kepler Sticka-Jones\\",\\"url\\":\\"https://keplersj.com\\"}}]}}",
      "type": "application/ld+json",
    },
  ],
  "styleTags": Array [],
  "title": "Blog | ",
  "titleAttributes": Object {},
}
`;
